src/components/Lesson/LessonList/index.tsx

```diff
@@ -1,43 +1,48 @@
-import React, { FC } from 'react';
-import { FlatList, TouchableOpacity } from 'react-native-gesture-handler';
+import React, { FC, memo } from 'react';
+import { FlatList } from 'react-native-gesture-handler';
 import { HorizontalLessonCard } from '../HorizontalLessonCard';
 import { VerticalLessonCard } from '../VerticalLessonCard';
 import useConnect from './connect';
 import { HorizontalView, VerticalView } from './styles';
 import { Props } from './types';
 
-export const LessonList: FC<Props> = ({ categoryFlag }) => {
+export const LessonList: FC<Props> = ({ isFiltered, lessons }) => {
   const { handleNavigateToDetail } = useConnect();
+  const categoryFlag = 'All'.localeCompare(isFiltered) === 0;
```

Comment:
Don't use localCompare. It's **never** necessary on typescript.
Also, isFiltered is a String, not a boolean. 

```suggestion
  const isFiltered = filter !== 'All';
```